internal/objects/multiclusteringress.go

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright 2019-2020 VMware, Inc.
+ * All Rights Reserved.
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*   http://www.apache.org/licenses/LICENSE-2.0
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+package objects
+
+import (
+	"sync"
+
+	"github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/pkg/utils"
+)
+
+//This package gives relationship APIs to manage a multi-cluster ingress object.
+
+var mciSvcListerInstance *SvcLister
+var once sync.Once
+
+func SharedMultiClusterIngressSvcLister() *SvcLister {
+	once.Do(func() {
+		mciSvcListerInstance = &SvcLister{
+			svcIngStore:         NewObjectStore(),
```

Comment:
Do we need svc to Ingress store, or is mci - svc import store sufficient ?

Reply:
yes, we need it. Since the function `GetSvcLister` returns `SvcLister`, I thought of reusing the functions and caches. svc name will be a combination of cluster, namespace, and service name and ingress name will be the mci object name.

Reply:
some of the caches are not being used, I am cleaning those up and doing sanity now.